app/react/V2/Routes/Settings/Thesauri/components/TableComponents.tsx

```diff
@@ -75,23 +68,30 @@ interface TableThesaurusValue extends ClientThesaurusValue, Omit<ClientThesaurus
   }[];
 }
 
+interface ThesaurusRow extends Omit<ClientThesaurusValue, 'values'> {
+  id?: string;
+  rowId: string;
+  label: string;
+  subRows?: Omit<ThesaurusRow, 'subRows'>[];
+  groupId?: string;
+}
+
 const columnHelper = createColumnHelper<any>();
-const columns = (actions: { edit: Function }) => [
+const columns = (actions: { edit?: Function }, thesaurus: ClientThesaurus) => [
   columnHelper.accessor('label', {
     id: 'label',
     header: LabelHeader,
     cell: ThesaurusValueLabel,
-    meta: { headerClassName: 'w-11/12' },
-  }) as ColumnDef<TableThesaurusValue, 'label'>,
+    meta: { headerClassName: 'w-11/12', data: { thesaurus } },
+  }) as ColumnDef<ThesaurusRow, 'label'>,
```

Comment:
The type should be inferred automatically. No need to cast it maybe?

Reply:
Totally!